include/envoy/http/filter.h

```diff
@@ -247,6 +255,22 @@ class StreamFilterBase {
 class StreamDecoderFilter : public StreamFilterBase {
 public:
   /**
+   * This routine is called on filter creation, setting the buffer limit for the
+   * filter.  Filters should abide by these limits unless custom configuration
+   * overrides the limit.  A buffer limit of 0 bytes indicates no limits are applied.
+   *
+   * If filters buffer enough bytes to hit the high watermark, it should either
+   * call on[Encoder|Decoder]FilterAboveWriteBufferHighWatermark to halt the
+   * flow of data or send an error response such as 413 (Payload Too Large).
+   *
+   * If the filter will return StopIterationAndBuffer it may override the buffer
+   * limit in settings, or the filter type.
+   *
+   * @param settings supplies the default buffer limit and filter type for this filter.
+   */
+  virtual void setDecoderBufferLimit(BufferLimitSettings& settings) PURE;
```

Comment:
thought: It might be better to just return a struct here to make it more clear that the filter needs to tell the connection manager what it wants. We could even make the default constructor for the struct set sane defaults. I think for this type of struct the compiler will optimize well.